sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -366,13 +372,9 @@ def create_entity(self, entity: EntityType, **kwargs) -> Dict[str, Any]:
             )
         except HttpResponseError as error:
             decoded = _decode_error(error.response, error.message)
-            if decoded.error_code == "PropertiesNeedValue":
-                if entity.get("PartitionKey") is None:
-                    raise ValueError("PartitionKey must be present in an entity") from error
-                if entity.get("RowKey") is None:
-                    raise ValueError("RowKey must be present in an entity") from error
+            _validate_key_values(decoded, entity.get("PartitionKey"), entity.get("RowKey"))
             _validate_tablename_error(decoded, self.table_name)
-            _reraise_error(error)
+            raise decoded from error
```

Comment:
It looks like we were originally just raising `error` rather than `decoded_error`.
While your change is probably correct... can we revert this to just `raise` for now? Let's also add an inline comment saying that we probably should have been raising `decoded_error` and to see if we can do it as a non-breaking change...